### DISCUSSION TITLE
Connecting to Buhler RM20 early machines

### DISCUSSION THREAD

**Discussion by oakroaster** [artisan-roaster-scope/artisan#1665] [Q&A]
Any ideas how to connect to Buhler RM20 early machines? They have siemens S7 313 cpu with no ethernet port. Only profibus connection to TP900 HMI.

**Comment by MAKOMO** [comment]
Artisan is not implementing the PROFIBUS protocol yet. Why not volunteer and add PROFIBUS support to Artisan? Maybe the [pyprofibus](https://pypi.org/project/pyprofibus/) Python lib can be help you on that implementation. Of course one might need also additional information from Buhler about the firmware in those machines and how they communicate exactly.
